### PR TITLE
Add gemrb - open source implementation of infinity engine

### PR DESCRIPTION
**Pull request by gizmo98**
Tested with Baldur’s Gate and Baldurs Gate 2. Based on:
https://github.com/amigagamer/gemrb-retropie/blob/master/gemrb.sh

https://github.com/RetroPie/RetroPie-Setup/issues/794

The content of the windows installation folder of one of these games must be copied to $romdir/ports/$gamename.

### CODE REVIEW

**Comment by HerbFargus** [comment]
I don't have the original game but it compiles properly and the configs all look in order. I don't think it likes the shareware version (or perhas mine was just too old)

**Comment by gizmo98** [comment]
If you use bg2 demo config must be modified:
http://www.raspians.com/Knowledgebase/gemrb-baldurs-gate-on-the-raspberry-pi/

File: scriptmodules/ports/gemrb.sh [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+rp_module_desc="gemrb - open-source implementation of Infinity Engine"
+rp_module_menus="4+"
+rp_module_flags=""
+
+function depends_gemrb() {
+    getDepends python-dev libopenal-dev 
+}
+
+function sources_gemrb() {
+    gitPullOrClone "$md_build" git://github.com/gemrb/gemrb.git
```
Comment (joolswills) [inline scriptmodules/ports/gemrb.sh]: please use https:// not the old deprecated git://

File: scriptmodules/ports/gemrb.sh [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+# at https://raw.githubusercontent.com/RetroPie/RetroPie-Setup/master/LICENSE.md
+#
+
+rp_module_id="gemrb"
+rp_module_desc="gemrb - open-source implementation of Infinity Engine"
+rp_module_menus="4+"
+rp_module_flags=""
+
+function depends_gemrb() {
+    getDepends python-dev libopenal-dev
```
Comment (joolswills) [inline scriptmodules/ports/gemrb.sh]: missing dependencies for cmake (i assume it needs sdl also ?)

**Comment by joolswills** [comment]
couple of minor things, but looks good - please fix and I will merge.

File: scriptmodules/ports/gemrb.sh [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+    addPort "$md_id" "Icewind2" "Icewind Dale 2" "$md_inst/bin/gemrb -C $configdir/Icewind2/GemRB.cfg"
+    addPort "$md_id" "Planescape" "Planescape Torment" "$md_inst/bin/gemrb -C $configdir/Planescape/GemRB.cfg"
+
+    mkRomDir "ports/baldurs1"
+    mkRomDir "ports/baldurs2"
+    mkRomDir "ports/icewind1"
+    mkRomDir "ports/icewind2"
+    mkRomDir "ports/planescape"
+    mkRomDir "ports/cache"
+
```
Comment (joolswills) [inline scriptmodules/ports/gemrb.sh]: mkRomDir should be before the addPort (put them last).

**Comment by gizmo98** [comment]
Thank you. I modified the noted sections.

**Comment by joolswills** [comment]
thanks.

**Pull request merged**